// @capgo/capacitor-navigation-bar/dist/plugin.cjs.js
'use strict';

var core = require('@capacitor/core');

/**
 * Predefined navigation bar colors.
 *
 * @since 1.0.0
 */
exports.NavigationBarColor = void 0;
(function (NavigationBarColor) {
    /** White color */
    NavigationBarColor["WHITE"] = "#FFFFFF";
    /** Black color */
    NavigationBarColor["BLACK"] = "#000000";
    /** Transparent color */
    NavigationBarColor["TRANSPARENT"] = "transparent";
})(exports.NavigationBarColor || (exports.NavigationBarColor = {}));

const NavigationBar = core.registerPlugin('NavigationBar', {
    web: () => Promise.resolve().then(function () { return web; }).then((m) => new m.NavigationBarWeb()),
});

class NavigationBarWeb extends core.WebPlugin {
    async setNavigationBarColor(options) {
        console.log('Cannot setNavigationBarColor on web', options);
        return;
    }
    async getNavigationBarColor() {
        console.log('Cannot getNavigationBarColor on web');
        return { color: '#000000', darkButtons: true };
    }
    async getPluginVersion() {
        return { version: 'web' };
    }
}

var web = /*#__PURE__*/Object.freeze({
    __proto__: null,
    NavigationBarWeb: NavigationBarWeb
});

exports.NavigationBar = NavigationBar;
